### PR TITLE
Added recipe for libdap4

### PR DESCRIPTION
**Pull request by aleksandervines**
A C++ SDK which contains an implementation of DAP2 and DAP4. This includes both Client- and Server-side support classes.

We want to provide this primarily to be able to provide dap support in gdal, where a contriubtion to the gdal feedstock would come next when this has been approved.

There is no support for this library on the Windows platform.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@aleksandervines Awesome contribution!

Note that due to a hiccup in Travis-CI you are pulling some extra recipes in there. We will need you to re-base once we fix the #900

**Comment by aleksandervines** [comment]
That is fine, I will await the fix.

**Comment by aleksandervines** [comment]
Hmm, I am doing something wrong, or is it supposed to check the other recipes, python-snappy, redis-py and singledispatch? :s

**Comment by ocefpaf** [comment]
@aleksandervines that is not your fault. Travis-CI is awfully slow these days and our feedstocking system relies on it to process the recipes and remove them. You can delete the other recipes here so the CIs will run better in your PR.

**Comment by ocefpaf** [comment]
(Ideally we should rebase this before merging once those recipe are cleared.)

File: recipes/libdap4/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+requirements:
+  build:
+    - automake
+    - autoconf
+    - libtool
+    - flex
+    - bison
+    - libgcc
```
Comment (ocefpaf) [inline recipes/libdap4/meta.yaml]: Do you need `libgcc` at build time? Usually we only add `defaults` `libgcc` at run time if we used `defaults` `gcc` at build time.

**Pull request closed without merging**